app/page.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { socket } from "./socket";
+
+export default function Page() {
+  const [isConnected, setIsConnected] = useState(false);
+  const [transport, setTransport] = useState("N/A");
+  const [text, setText] = useState("");
+  const [messages, setMessages] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (socket.connected) {
+      onConnect();
+    }
+
+    function onConnect() {
+      setIsConnected(true);
+      setTransport(socket.io.engine.transport.name);
+
+      socket.io.engine.on("upgrade", (transport) => {
+        setTransport(transport.name);
+      });
+    }
+
+    function onDisconnect() {
+      setIsConnected(false);
+      setTransport("N/A");
+    }
+
+    function onMessage(message: string) {
+      setMessages((messages) => [...messages, message]);
+    }
+
+    socket.on("connect", onConnect);
+    socket.on("disconnect", onDisconnect);
+    socket.on("message", onMessage);
+
+    return () => {
+      socket.off("connect", onConnect);
+      socket.off("disconnect", onDisconnect);
+      socket.off("message", onMessage);
+    };
+  }, []);
```

Comment:
I would extract the socket.io stuff into it's own hook. also see [Socket.IO | Typescript](https://socket.io/docs/v4/typescript/) for properly defining the types we expect between the frontend and server code.